src/main/java/com/google/devtools/build/lib/skyframe/PrerequisiteProducer.java

```diff
@@ -405,6 +429,29 @@ public boolean evaluate(
     return true;
   }
 
+  private PlatformInfo loadTargetPlatformInfo(Environment env)
+      throws InterruptedException, ToolchainException {
+    PlatformConfiguration platformConfiguration = targetAndConfiguration.getConfiguration()
+        .getFragment(PlatformConfiguration.class);
+    if (platformConfiguration == null) {
+      return null;
```

Comment:
In line 329, I check `env.valuesMissing()` rather than the return type of this function. I would have expected that to prevent the restart in this case. Do you think that's correct?